subsys/net/lib/download_client/src/transports/http.c

```diff
@@ -0,0 +1,577 @@
+/*
```

Comment:
Moving my previous comment here to be able to have a discussion.

Looks like there's some problem with P-GPS. The download sometimes fails:
```
mosh:~$ gnss pgps enable
GNSS: Getting P-GPS predictions from nRF Cloud...
Modem domain event: CE-level 0
RRC mode: Connected
[00:00:34.508,728] <inf> nrf_cloud_pgps: Storage base:0xEB000, size:86016
[00:00:34.509,399] <inf> nrf_cloud_pgps: Checking P-GPS header: Schema version:0, type:0, num:0, count:0
[00:00:34.509,429] <inf> nrf_cloud_pgps:   size:0, period (minutes):0, GPS day:0, GPS time:0
[00:00:34.509,460] <wrn> nrf_cloud_pgps: One or more fields are wrong
GNSS: Processing P-GPS response
GNSS: P-GPS response processed
GNSS: P-GPS predictions requested
[00:00:35.786,682] <wrn> download_client: Protocol not specified for pgps.nrfcloud.com/public/16377-50400_16384-36000.bin, attempting http
[00:00:35.791,137] <inf> nrf_cloud_pgps: Searching for prediction
[00:00:35.791,168] <inf> nrf_cloud_pgps: Predictions not loaded yet
[00:00:35.838,592] <inf> download_client: Connecting to 2600:9000:2368:5a00:3:e9a6:59c0:93a1
[00:00:35.987,060] <inf> download_client: Downloaded 44/84268 bytes (0%)
[00:00:35.987,091] <inf> nrf_cloud_pgps: Checking P-GPS header: Schema version:104, type:116, num:28788, count:12591
[00:00:35.987,091] <inf> nrf_cloud_pgps:   size:12590, period (minutes):12832, GPS day:13872, GPS time:1918988320
[00:00:35.987,121] <wrn> nrf_cloud_pgps: One or more fields are wrong
[00:00:35.987,915] <err> nrf_cloud_pgps: Download failed: -22
```
This was with modem_shell, but I also saw error `-22` with the Location sample. This does not happen always, but when it has failed, the first downloaded chunk has been very small, in this case 44 bytes. This might or might not be related to the issue.

Also, looks like the URI is missing the protocol, causing this warning:
`[00:00:35.786,682] <wrn> download_client: Protocol not specified for pgps.nrfcloud.com/public/16377-50400_16384-36000.bin, attempting http`

Reply:
After further tests, the number of downloaded bytes seems to always be 44 when the failure happens. nRF Cloud P-GPS fails, because the P-GPS header parsed from the received data is invalid. I dumped the received data and instead of P-GPS data there's this:
```
http/1.1 206 partial content
content-type:
```

Reply:
This is likely related to the issue with SLM. I saw similar data when there was attempt to compare to mcuboot hash to verify the image type.

Reply:
Yes, I think this is the same issue as @trantanen was seeing. I found the issue and will push the fix now.

Reply:
Oh.. This used to be a problem already a year ago.
Was fixed in https://github.com/nrfconnect/sdk-nrf/pull/13814 so are we introducing same regression again?

Library should tolerate this and thread HTTP as a stream. Not every `recv()` contains enough data to be a valid block.

Reply:
@eivindj-nordic One additional thing is that P-GPS now seems to use HTTP, not HTTPS, because it does not explicitly specify the transport and this defaults to HTTP. I think the default should be HTTPS when sec_tags are provided and HTTP when not.

Reply:
@SeppoTakalo Could you elaborate what the problem was at that point? Now we are sending what we recv  (except the http header) directly to the application. Whether it is 1 byte or a full buffer.

Reply:
@trantanen Agreed, the http transport should use https when a sec_tags are provided already, though the warning will say `... attempting http`.